wtransport/src/endpoint.rs

```diff
@@ -564,18 +565,66 @@ where
 type DynFutureIncomingSession =
     dyn Future<Output = Result<SessionRequest, ConnectionError>> + Send + Sync;
 
-/// [`Future`] for an in-progress incoming connection attempt.
+/// [`IntoFuture`] for an in-progress incoming connection attempt.
 ///
 /// Created by [`Endpoint::accept`].
-pub struct IncomingSession(Pin<Box<DynFutureIncomingSession>>);
+pub struct IncomingSession(quinn::Incoming);
```

Comment:
I believe there is a good reason to do it this way, since the inner type is very different. If they used the same type, it would probably have to be an `enum` with two variants. Before the first `await`, you can could call the methods. After `await`, the methods would panic because the `quinn::Incoming` would have already been converted into the `Pin<Box<DynFutureIncomingSession>>`.